python/lsst/pipe/base/graphBuilder.py

```diff
@@ -845,6 +881,11 @@ def resolveDatasetRefs(
                         collectionTypes=CollectionType.RUN,
                     )
 
+        idMaker: Optional[_DatasetIdMaker] = None
+        if resolveRefs:
+            assert run is not None, "run cannot be None when resolveRefs is True"
```

Comment:
Is assert really the right kind of exception to raise here?

Reply:
I think it's fine here, it is internal code only called from `makeGraph()`. `makeGraph()` has an explicit check that raises `ValueError`, here assert is mostly to keep `mypy` happy.